includes/admin/class-sensei-learner-management.php

```diff
@@ -272,12 +273,56 @@ public function load_data_object( $name = '', $data = 0, $optional_data = null )
 	 * @access public
 	 */
 	public function learners_page() {
-		$type = isset( $_GET['view'] ) ? esc_html( $_GET['view'] ) : false;
-		if ( $this->bulk_actions_controller->get_view() === $type ) {
-			$this->bulk_actions_controller->learner_admin_page();
-			return;
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		if ( ! empty( $_GET['course_id'] ) ) {
+			$this->output_course_page();
+		} else {
+			$this->output_main_page();
 		}
 
+	}
+
+	/**
+	 * Display the students main page.
+	 *
+	 * @since x.x.x
+	 */
+	private function output_main_page() {
+		// Load Learners data.
+		$sensei_learners_main_view = new Sensei_Learners_Admin_Bulk_Actions_View( $this->bulk_actions_controller, $this );
+		$sensei_learners_main_view->prepare_items();
+
+		// Wrappers.
+		do_action( 'sensei_learner_admin_before_container' );
+		?>
+		<div id="woothemes-sensei" class="wrap woothemes-sensei">
+		<?php
+		do_action( 'sensei_learner_admin_wrapper_container', 'top' );
+		$sensei_learners_main_view->output_headers();
+		?>
+		<div id="poststuff" class="sensei-learners-wrap">
+			<div class="sensei-learners-main">
+				<?php $sensei_learners_main_view->display(); ?>
+			</div>
+			<div class="sensei-learners-extra">
+				<?php do_action( 'sensei_learner_admin_extra' ); ?>
+			</div>
+		</div>
+		<?php
+		do_action( 'sensei_learner_admin_wrapper_container', 'bottom' );
+		?>
+		</div>
+		<?php
+		do_action( 'sensei_learner_admin_after_container' );
+	}
```

Comment:
This is a copy of `Sensei_Learners_Admin_Bulk_Actions_Controller::learner_admin_page`. Probably would need to be refactored/updated or combined with `output_course_page` in another PR.

Reply:
Good idea! Updated in c68e513d5b0aa2aa29c10ff71c6d3a8127a10ff0.